jupyterlab_hdf/util.py

```diff
@@ -293,3 +301,10 @@ def uriJoin(*parts):
 
 def uriName(uri):
     return uri.split('/')[-1]
+
+## External link that can be parsed by hobjDict methods
+class HobjExternalLink:
+    def __init__(self, link, name) -> None:
+        self.name = name
+        self.original_link = link
+        self.attrs = []
```

Comment:
`h5py.ExternalLink` has only two attributes: `filename` and `path` that is the uri in the external file. I use this wrapper to match the signature of `h5py.Dataset` and `h5py.Group`.

Reply:
we should probably just use `h5py.ExternalLink`. What are the `attrs` needed here for exactly? Fundamentally, there's no particular reason why the sig of the link type should match that of `Dataset` and `Group`

Reply:
True for `attrs`.

However the problem with h5py.ExternalLink is that there is no way to get back to the uri in the original file (`name`) that is used to construct the paths.

Reply:
I see the problem. Still, I think that creating our own `HobjExternalLink` has a bit of a smell.

So I guess for now leave `HobjExternalLink` in but get rid of `.attrs`. I'll do a followup PR to clean this up as well as some other related sloppy code in `util.py` (ie all of the `hobjFoo` functions, which generate the query returns)